packages/contentstack-migration/src/utils/modules.js

```diff
@@ -0,0 +1,172 @@
+const fs = require('fs');
+const { execSync } = require('child_process');
+const path = require('path');
+const { sanitizePath } = require('@contentstack/cli-utilities');
+const os = require('os');
+
+const internalModules = new Set([
+  'assert',
+  'buffer',
+  'child_process',
+  'cluster',
+  'console',
+  'crypto',
+  'dgram',
+  'dns',
+  'domain',
+  'events',
+  'fs',
+  'http',
+  'http2',
+  'https',
+  'inspector',
+  'module',
+  'net',
+  'os',
+  'path',
+  'perf_hooks',
+  'process',
+  'punycode',
+  'querystring',
+  'readline',
+  'repl',
+  'stream',
+  'string_decoder',
+  'timers',
+  'tls',
+  'trace_events',
+  'tty',
+  'url',
+  'util',
+  'v8',
+  'vm',
+  'wasi',
+  'worker_threads',
+  'zlib',
+]);
+
+function checkWritePermissionToDirectory(directory) {
+  try {
+    fs.accessSync(directory, fs.constants.W_OK);
+    return true;
+  } catch (err) {
+    console.log(`You don't have write access to directory`);
+    return false;
+  }
+}
+
+function doesPackageJsonExist(directory) {
+  return fs.existsSync(path.join(sanitizePath(directory), 'package.json'));
+}
+
+function scanDirectory(directory) {
+  return fs.readdirSync(directory);
+}
+
+function scanFileForDependencies(directory, files) {
+  const dependencies = new Set();
+
+  files.forEach((file) => {
+    const filePath = path.join(sanitizePath(directory), sanitizePath(file));
+    if (path.extname(filePath) === '.js') {
+      const fileContent = fs.readFileSync(filePath, 'utf-8');
+      findModulesSync(fileContent).forEach((dep) => dependencies.add(dep));
+    }
+  });
+
+  return [...dependencies];
+}
+
+function createPackageJson(directory) {
+  const templateString = `{
+    "name": "MigrationPackage",
+    "version": "1.0.0",
+    "main": "1.js",
```

Comment:
@cs-raj , any specific reason for mentioning 1.js in main.

Reply:
No, It was just the file with which it was added, shall I remove it?

Reply:
Removed